pkg/drivers/cdp/driver.go

```diff
@@ -135,6 +140,131 @@ func (drv *Driver) Open(ctx context.Context, params drivers.Params) (drivers.HTM
 	return LoadHTMLPage(ctx, conn, params)
 }
 
+func (drv *Driver) Parse(ctx context.Context, content []byte) (drivers.HTMLPage, error) {
+	logger := logging.FromContext(ctx)
+
+	err := drv.init(ctx)
+
+	if err != nil {
+		logger.
+			Error().
+			Timestamp().
+			Err(err).
+			Str("driver", drv.options.Name).
+			Msg("failed to initialize the driver")
+
+		return nil, err
+	}
+
+	// Args for a new target belonging to the browser context
+	createTargetArgs := target.NewCreateTargetArgs(BlankPageURL)
+
+	// Set it to incognito mode
+	createTargetArgs.SetBrowserContextID(drv.contextID)
+
+	// New target
+	createTarget, err := drv.client.Target.CreateTarget(ctx, createTargetArgs)
+
+	if err != nil {
+		logger.
+			Error().
+			Timestamp().
+			Err(err).
+			Str("driver", drv.options.Name).
+			Msg("failed to create a browser target")
+
+		return nil, err
+	}
+
+	// Connect to target using the existing websocket connection.
+	conn, err := drv.session.Dial(ctx, createTarget.TargetID)
+
+	if err != nil {
+		logger.
+			Error().
+			Timestamp().
+			Err(err).
+			Str("driver", drv.options.Name).
+			Msg("failed to establish a connection")
+
+		return nil, err
+	}
+
+	// New client
+	client := cdp.NewClient(conn)
+
+	// Configure client
+	err = runBatch(
+		func() error {
+			return client.Page.SetLifecycleEventsEnabled(
+				ctx,
+				page.NewSetLifecycleEventsEnabledArgs(true),
+			)
+		},
+
+		func() error {
+			return client.DOM.Enable(ctx)
+		},
+
+		func() error {
+			return client.Runtime.Enable(ctx)
+		},
+
+		func() error {
+			return client.Network.Enable(ctx, network.NewEnableArgs())
+		},
+
+		func() error {
+			return client.Page.SetBypassCSP(ctx, page.NewSetBypassCSPArgs(true))
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Get frame tree
+	ft, err := client.Page.GetFrameTree(ctx)
```

Comment:
I guess we need to invoke this method after setting a document's content.

Reply:
SetDocumentContent(...) takes the frameID as an argument, so it is called before to get that.

Reply:
Ok, thanks for explanation!

Reply:
Write a comment with explanation from this thread, please